extensions/amp-consent/0.1/consent-state-manager.js

```diff
@@ -14,76 +14,181 @@
  * limitations under the License.
  */
 
+import {Observable} from '../../../src/observable';
 import {Services} from '../../../src/services';
+import {dev} from '../../../src/log';
+
+const TAG = 'CONSENT-STATE-MANAGER';
+
+/**
+ * @enum {number}
+ */
+export const CONSENT_ITEM_STATE = {
+  UNKNOWN: 0,
+  GRANTED: 1,
+  REJECTED: 2,
+};
 
 export class ConsentStateManager {
   constructor(ampdoc) {
-    this.ampdoc = ampdoc;
+    /** @private {?../../../src/service/ampdoc-impl.AmpDoc} */
+    this.ampdoc_ = ampdoc;
 
+    /** @private {!Object<string, ConsentInstance>} */
     this.instances_ = {};
+
+    /** @private {!Object<string, ?Observable>}*/
+    this.consentChangeObservable_ = {};
+
+    /** @private {!Object<string, ?function()>} */
+    this.consentReadyResolver_ = {};
+
+    /** @private {!Object<string, ?Promise>} */
+    this.consentReadyPromise_ = {};
   }
 
   /**
    * Register and create a consent instance to manage state
    * @param {string} instanceId
    */
   registerConsentInstance(instanceId) {
-    this.instances_[instanceId] = new ConsentInstance(this.ampdoc, instanceId);
+    dev().assert(!this.instances_[instanceId],
+        `${TAG}: instance already registered`);
+    this.instances_[instanceId] = new ConsentInstance(this.ampdoc_, instanceId);
+    this.consentChangeObservable_[instanceId] = new Observable();
+    if (this.consentReadyResolver_[instanceId]) {
+      this.consentReadyResolver_();
+      this.consentReadyPromise_[instanceId] = null;
+      this.consentReadyResolver_[instanceId] = null;
+    }
+  }
+
+  /**
+   * Ignore a consent instance.
+   * @param {string} instanceId
+   */
+  ignoreConsentInstance(instanceId) {
+    dev().assert(this.instances_[instanceId],
+        `${TAG}: cannot find this instance`);
+    this.consentChangeObservable_[instanceId].fire(CONSENT_ITEM_STATE.GRANTED);
```

Comment:
new State should be used

Reply:
Added TODO. Will address in following PR.